Clarify contribution guide on integration tests (#8718)
ansible-collections/community.general: CONTRIBUTING.md

The following commands show how to run integration tests:

> #### In docker

Reviewer: ```suggestion
### In Docker
```

```.bash
# Run integration tests in a Docker container for a specific module, e.g. community.general.pacman:

> ansible-test integration -v --docker default pacman

Reviewer: The default container is almost **never** a good idea to use for integration tests.

**Especially** for the pacman module, there's exactly one single container that's useful (and that's not even included in ansible-test's list of containers). So this example is not very helpful :)

# Integration test on a specific docker image. Find out which target docker images are supported by

> # running ansible-test integration --help and choose one, e.g. fedora35:

Reviewer: ```suggestion
# running "ansible-test integration --help" and choose one, for example fedora35:
```

> ansible-test integration -v --docker fedora35

Reviewer: I would update this to `fedora40`, since `fedora35` is already pretty old.

Author: The reason I left is as-is, is because no matter the value we update this to, it will get out-of-date. My reasoning is that specifying an older version might make contributors more likely to halt before copy-pasting the command. I was even debating setting it to ubuntu1004 just to drive home that the command is not copy-pastable.

If you do not agree, I can change it.

Reviewer: I would bump it to the latest when we change the text anyway, to reduce the amount of friction for folks who simply copy'n'paste.

```

> #### Without isolation

Reviewer: ```suggestion
### Without isolation
```

ansible-test units --docker -v --python 3.8 tests/unit/plugins/modules/net_tools/test_nmcli.py
```

### Integration tests

The following commands show how to run integration tests:

#### In docker

> Integration tests on docker require two parameters:

Reviewer: The product is called Docker (uppercase), not docker (lowercase). (That Docker's logo uses the word 'docker' in lower-case doesn't mean the product is called 'docker'.)
```suggestion
Integration tests on Docker require two parameters:
```
Also this statement itself is not correct, the test name isn't required.

Reviewer: No, that looks like a bug in ansible-test, which we can also fix on our side by removing the invalid `cloud/gandi` entry in the aliases file of that test (I will do that in a minute).

Author: I'll rebase this PR on main and update the wording

> - `image_name`: The name of the docker image. To get the list of supported docker images, run
>   `ansible-test integration --help` and look for _target docker images_.

Reviewer: ```suggestion
- `image_name`: The name of the Docker image. To get the list of supported Docker images, run
  `ansible-test integration --help` and look for _target docker images_.
```

> - `module_name`: The name of the module. For example, pacman in case of community.general.pacma

Reviewer: ```suggestion
- `test_name`: The name of the integration test. For modules, this equals the short name of the module; for example, `pacman` in case of `community.general.pacman`. For plugins, the plugin type is added before the plugin's short name, for example `callback_yaml` for the `community.general.yaml` callback.
```
(Probably should have line breaks.)

```.bash
# Template
ansible-test integration -v --docker image_name module_name

> # Example community.general.pacman module on fedora35 docker image:

Reviewer: ```suggestion
# Example community.general.pacman module on fedora35 Docker image:
```
